docs/09-environment-integrations/nethack.md

```diff
@@ -0,0 +1,69 @@
+# NetHack
+<video width="800" controls autoplay><source src="https://github.com/BartekCupial/sample-factory/assets/92169405/47884b73-beeb-4303-a72f-75d202aa87a8" type="video/mp4"></video>
+## Installation
+To install NetHack, you need nle and its dependencies.
+
+```bash
+# nle dependencies
+apt-get install build-essential python3-dev python3-pip python3-numpy autoconf libtool pkg-config libbz2-dev
+conda install cmake flex bison lit
+
+# install nle locally and modify it to enable seeding and handle rendering with gymnasium
+git clone https://github.com/facebookresearch/nle.git nle && cd nle \
+&& git checkout v0.9.0 && git submodule init && git submodule update --recursive \
+&& sed '/#define NLE_ALLOW_SEEDING 1/i#define NLE_ALLOW_SEEDING 1' include/nleobs.h -i \
+&& sed '/self\.nethack\.set_initial_seeds = f/d' nle/env/tasks.py -i \
+&& sed '/self\.nethack\.set_current_seeds = f/d' nle/env/tasks.py -i \
+&& sed '/self\.nethack\.get_current_seeds = f/d' nle/env/tasks.py -i \
+&& sed '/def seed(self, core=None, disp=None, reseed=True):/d' nle/env/tasks.py -i \
+&& sed '/raise RuntimeError("NetHackChallenge doesn.t allow seed changes")/d' nle/env/tasks.py -i \
+&& sed -i '/def render(self, mode="human"):/a\        if not self.last_observation:\n            return' nle/env/base.py \
+&& python setup.py install && cd .. 
```

Comment:
Had issues with this in Python 3.11 conda env (pybind11 errors), but worked fine in 3.10.

Can we add a note that 3.10 is recommended and there are known issues in 3.11?

Reply:
Actually in setup.py we have `python_requires=">=3.8",`, but at the same time we use features from python 3.10 (typehints `Tensor | float`). Maybe we should bump python_requires to 3.10.

Reply:
Added some info in docs about issues with python 3.11 and above.

Reply:
> Actually in setup.py we have `python_requires=">=3.8",`, but at the same time we use features from python 3.10 (typehints `Tensor | float`). Maybe we should bump python_requires to 3.10.

We run unit tests on 3.8, 3.9, and 3.10 and all currently pass.
`Tensor | float` work because there's some import "from future import annotations" or smth like that

Reply:
I guess I forgot to use that import in some parts of my other code. Thanks :)